dlt/destinations/synapse/synapse.py

```diff
@@ -0,0 +1,142 @@
+
+from dlt.destinations.sql_jobs import SqlStagingCopyJob, SqlMergeJob
+
+from dlt.destinations.insert_job_client import InsertValuesJobClient
+
+from dlt.destinations.synapse import capabilities
+from dlt.destinations.synapse.sql_client import PyOdbcSynapseClient
+from dlt.destinations.synapse.configuration import SynapseClientConfiguration
+from dlt.destinations.sql_client import SqlClientBase
+
+
+SCT_TO_PGT: Dict[TDataType, str] = {
+    "complex": "nvarchar(4000)",
+    "text": "nvarchar(4000)",
+    "ntext": "nvarchar(4000)",
+    "double": "float",
+    "bool": "bit",
+    "timestamp": "datetimeoffset",
+    "date": "date",
+    "bigint": "bigint",
+    "binary": "varbinary(8000)",
+    "decimal": "decimal(%i,%i)",
+    "time": "time"
+}
+
+PGT_TO_SCT: Dict[str, TDataType] = {
+    "nvarchar": "text",
+    "float": "double",
+    "bit": "bool",
+    "datetimeoffset": "timestamp",
+    "date": "date",
+    "bigint": "bigint",
+    "varbinary": "binary",
+    "decimal": "decimal",
+    "time": "time"
+}
+
+HINT_TO_SYNAPSE_ATTR: Dict[TColumnHint, str] = {
+    "unique": "UNIQUE"
+}
+
+class SynapseStagingCopyJob(SqlStagingCopyJob):
+
+    @classmethod
+    def generate_sql(cls, table_chain: Sequence[TTableSchema], sql_client: SqlClientBase[Any]) -> List[str]:
+        sql: List[str] = []
+        for table in table_chain:
+            with sql_client.with_staging_dataset(staging=True):
+                staging_table_name = sql_client.make_qualified_table_name(table["name"])
+            table_name = sql_client.make_qualified_table_name(table["name"])
+            # drop destination table
+            sql.append(f"DROP TABLE {table_name};")
+            # moving staging table to destination schema
+            sql.append(f"ALTER SCHEMA {sql_client.fully_qualified_dataset_name()} TRANSFER {staging_table_name};")
+            # recreate staging table
+            sql.append(f"SELECT * INTO {staging_table_name} FROM {table_name} WHERE 1 = 0;")
+        return sql
+
+
+class SynapseMergeJob(SqlMergeJob):
+    @classmethod
+    def gen_key_table_clauses(cls, root_table_name: str, staging_root_table_name: str, key_clauses: Sequence[str], for_delete: bool) -> List[str]:
+        """Generate sql clauses that may be used to select or delete rows in root table of destination dataset
+        """
+        if for_delete:
+            # MS SQL doesn't support alias in DELETE FROM
+            return [f"FROM {root_table_name} WHERE EXISTS (SELECT 1 FROM {staging_root_table_name} WHERE {' OR '.join([c.format(d=root_table_name,s=staging_root_table_name) for c in key_clauses])})"]
+        return SqlMergeJob.gen_key_table_clauses(root_table_name, staging_root_table_name, key_clauses, for_delete)
+
+    @classmethod
+    def _to_temp_table(cls, select_sql: str, temp_table_name: str) -> str:
+        return f"SELECT * INTO {temp_table_name} FROM ({select_sql}) as t;"
+
+    @classmethod
+    def _new_temp_table_name(cls, name_prefix: str) -> str:
```

Comment:
shouldn't there be a uniq id included in the name to avoid collissions? See the base SqlMergeJob Class

Reply:
`_new_temp_table_name` is calling the base class function in the line `name = SqlMergeJob._new_temp_table_name(name_prefix)`, so unique_ids are being used. 

Should we instead just be calling the base function directly?